File: backend/AdSetLeadDesafio/DAL/DTOs/VeiculoEnvioDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.DTOs
{
    public class VeiculoEnvioDTO
    {
        [Required]
        public string Marca { get; set; }
        [Required]
        public string Modelo { get; set; }
        [Required]
        public int Ano { get; set; }
        [Required]
        public string Placa { get; set; }
        [Required]
        public string Cor { get; set; }
        [Required]
        public double Preco { get; set; }
        public int Km { get; set; }
        public string Opcionais { get; set; }
    }
}

File: frontend/src/app/models/veiculoEnvioDTO.ts
export class VeiculoEnvioDTO {
    public placa?: string | null = '';
    public marca?: string | null = '';
    public modelo?: string | null = '';
    public ano?: number | null = null;
    public km?: number | null = null;
    public preco?: number | null = null;
    public opcionais?: string | null = '';
    public cor?: string | null = '';
}

export class VeiculoEnvioPacoteDTO {
    public id?: number
    public pacoteICarros?: number
    public pacoteWebMotors?: number
}